scipy/_lib/_util.py

```diff
@@ -657,6 +657,33 @@ def __call__(self, x):
         return self.f(x, *self.args)
 
 
+class _ScalarFunctionWrapper:
```

Comment:
This class ensures that a scalar is returned from a user function. class is used to allow pickling.
It's added to `_lib` so that it can be used in several places without import problems.

A `_VectorFunctionWrapper` will be added in future PRs.